gcp-vertex-ai-mojo-scorer/src/main/java/ai/h2o/mojos/deploy/gcp/vertex/ai/controller/ModelsApiController.java

```diff
@@ -74,8 +85,40 @@ public static ScoreRequest getRestScoreRequest(
     request.setIdField(gcpRequest.getParameters().getIdField());
     request.setFields(gcpRequest.getParameters().getFields());
     
+    // Check if a pre-processing script was provided, if so, use it first
+    Map<String, String> env = System.getenv();
+    String preProccessingScript = env.getOrDefault("PREPROCESSING_SCRIPT_PATH", "");
+    
+    if (!preProccessingScript.isEmpty()) {
+      // Write request body to JSON file to be consumed by Python script
+      ObjectMapper mapper = new ObjectMapper();
+      
+      String fileName = UUID.randomUUID().toString();
+      mapper.writeValue(new File("/tmp/" + fileName), gcpRequest);
+      
+      // Run Python pre processing script and reload request body with updated data
+      ProcessBuilder processBuilder;
+      processBuilder = new ProcessBuilder("python", "/tmp/preprocessing_script.py", fileName);
+      processBuilder.redirectErrorStream(true);
+      log.info("Waiting for script to run");
+      Process process = processBuilder.start();
+      int exitCode = process.waitFor();
+      
+      if (exitCode != 0) {
+        throw new Exception("Preprocessing script failed.");
+      }
+      
+      // Read preprocessed data
+      JsonReader reader = new JsonReader(new FileReader("/tmp/" + fileName));
```

Comment:
I'm pretty sure this could be just a byte reader, although I don't know how much it actually changes things. If `JsonReader` is an extra dependency, I'd encourage to not use it. But if its a built in, then I don't care